src/Compilers/CSharp/Portable/Syntax/SyntaxNodeExtensions.cs

```diff
@@ -100,6 +100,71 @@ internal static bool IsValidScopeDesignator(this ExpressionSyntax expression)
             }
         }
 
+        internal static bool IsVariableDeclarationInitialization(this SyntaxNode node)
+        {
+            Debug.Assert(node != null);
+
+            SyntaxNode equalsValueClause = node.Parent;
+
+            if (!equalsValueClause.IsKind(SyntaxKind.EqualsValueClause))
+            {
+                return false;
+            }
+
+            SyntaxNode variableDeclarator = equalsValueClause.Parent;
+
+            if (!variableDeclarator.IsKind(SyntaxKind.VariableDeclarator))
+            {
+                return false;
+            }
+
+            return variableDeclarator.Parent.IsKind(SyntaxKind.VariableDeclaration);
+        }
+
+        /// <summary>
+        /// PROTOTYPE(span):
+        /// short work around until we look into spilling stackalloc expressions
+        /// Because the instruction cannot have any values on the stack before CLR execution.
+        /// Limit it to assignments and conditional expressions for now.
+        /// </summary>
+        internal static bool IsLegalSpanStackAllocPosition(this SyntaxNode node)
+        {
+            Debug.Assert(node != null);
+
+            while (node.Parent.IsKind(SyntaxKind.CastExpression))
+            {
+                node = node.Parent;
+            }
+
+            if (node.Parent.IsKind(SyntaxKind.ConditionalExpression))
```

Comment:
Yes. for example `ConditionalExpressionOnSpan_Convertible` in the semantic tests project. We won't allow nesting for now. This is a work around until we look into spilling for stackalloc instructions.

---
In reply to: [132792122](https://github.com/dotnet/roslyn/pull/21006#discussion_r132792122) [](ancestors = 132792122)